File: Chapter-2/Java/Problem3.java
//import required libraries

import java.util.Scanner;

public class Problem3 {

    public static void main(String[] args) {
        double L1, L2, G1, G2, d;
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter latitudes in two lines:");
        L1 = sc.nextDouble();
        L2 = sc.nextDouble();
        System.out.println("Enter longitudes in two lines:");
        G1 = sc.nextDouble();
        G2 = sc.nextDouble();
        d = 3963 * Math.acos(Math.sin(L1) * Math.sin(L2) + Math.cos(L1) * Math.cos(L2) * Math.cos(G2 - G1));
        System.out.printf("Latitudes: (%.3f,%.3f)", L1, L2);
        System.out.printf("Longitudes: (%.3f,%.3f)", G1, G2);
        System.out.printf("Distance in nautical miles: %.3f", d);
        sc.close();
    }
}

File: Chapter-2/Python/Problem3.py
import math as m
L1,L2 = list(map(float,input("Enter the latitudes:").split()))
G1,G2 = list(map(float,input("Enter the latitudes:").split()))
D = 3963*m.acos(m.sin(L1)*m.sin(L2) + m.cos(L1)*m.cos(L2)*m.cos(G2-G1))

print("Latitudes : (%.3f,%.3f)" %(L1,L2))
print("Longitudes : (%.3f,%.3f)" %(G1,G2))
print("Distance in nautical miles: %.3f" %(D))

